### PR TITLE
test(wallet)_: status-go integration tests

### LINKED ISSUE #15020: POC for testing status go
Acceptance criteria: ability to interact with status-go backend for desktop application without accessing client UI

Some context:

https://github.com/status-im/status-go/issues/4595
https://github.com/status-im/status-go/pull/4913

https://www.notion.so/Debugging-status-go-and-Waku-f1c516d37be741188530ac707d5b8f2e
https://www.notion.so/Telemetry-service-3d520063d3c24717b3f8823084f9509b
https://github.com/status-im/status-go/blob/7816ee186d27acf30b2d3d7ef217cfec7416efd6/services/ext/api.go#L452
https://www.notion.so/How-to-access-json-rpc-in-status-go-76c127efb9074b5daac8ba3663481073

https://www.notion.so/Postman-stuff-71bc0689db00476da55d31bbe602078d

### PR DESCRIPTION
**Pull request by antdanchenko**
MVP status-go integration tests

MVP of Integration tests for status-go which is [running as daemon](https://github.com/status-im/status-go/issues/5175) and uses [anvil](https://book.getfoundry.sh/reference/anvil/) as RPCURL with ChainID 31337, also status-im contract(s) are being deployed to the network

For more details please see [integration-tests/README.MD](https://github.com/status-im/status-go/blob/45be54008ed665f4441d5c86ff4b67c4a7b0b631/integration-tests/README.MD)

Important changes:
- [x] status-go will crash on start, fix is in https://github.com/status-im/status-go/pull/5301 and should be merged before trying to run this PR
- [x] only 2 contracts (status-network-token-v2; communities-contracts) are being deployed atm, more contracts will be added in future
- [x] changed builder version for _assets/build/Dockerfile to fix `/go/src/github.com/status-im/status-go/go.mod:5: unknown directive: toolchain`
- [x] please ignore files from integration-tests/schemas during review, they are used as expected result for test verifications  

Closes #
https://github.com/status-im/status-desktop/issues/15020
https://github.com/status-im/desktop-qa-automation/issues/445
https://github.com/status-im/desktop-qa-automation/issues/385

### CODE REVIEW

File: integration-tests/README.MD [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+
+## How to Install
+
+* Install [Docker](https://docs.docker.com/engine/install/) and [Docker Compose](https://docs.docker.com/compose/install/)
+- Checkout all listed contract repositories into one folder: 
+    - [Status Network Token V2](https://github.com/status-im/status-network-token-v2)
+    - [Communities Contracts](https://github.com/status-im/communities-contracts?tab=readme-ov-file#installation-and-development)
+* Add environment variable, path to contract repos e.g. `export PATH_TO_CONTRACT_REPOS=~/directory-with-every-contract-repo`
+* Install [Python 3.10.14](https://www.python.org/downloads/)
+* In `integration-tests/tests` directory, run `pip install -r requirements.txt`
```
Comment (anastasiyaig) [inline integration-tests/README.MD]: please commit the requirements file as it is not yet present

Comment (antdanchenko) [reply integration-tests/README.MD]: thanks, it is present in `integration-tests`, a directory above, will change the readme

File: integration-tests/README.MD
Comment (Samyoul) [inline integration-tests/README.MD:1]: Really nice documentation.

File: integration-tests/README.MD [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+    - [Status Network Token V2](https://github.com/status-im/status-network-token-v2)
+    - [Communities Contracts](https://github.com/status-im/communities-contracts?tab=readme-ov-file#installation-and-development)
+* Add environment variable, path to contract repos e.g. `export PATH_TO_CONTRACT_REPOS=~/directory-with-every-contract-repo`
+* Install [Python 3.10.14](https://www.python.org/downloads/)
+* In `integration-tests` directory, run `pip install -r requirements.txt`
+* **Optional (for test development)**: Use Python virtual environment for better dependency management. You can follow the guide [here](https://akrabat.com/creating-virtual-environments-with-pyenv/):
+
+## How to Run
+
+- In `integration-tests/tests` run `docker compose up --build`, as result:
```
Comment (Samyoul) [inline integration-tests/README.MD:26]: Do you think that there would be any value in explicitly telling the reader to navigate to `integration-tests/tests`?

Something like:

```suggestion
- Navigate to `integration-tests/tests`
- Run `docker compose up --build`, as result:
```

**Review by anastasiyaig — APPROVED** [review]
I was able to run this on my mac m2, but with some changes in the yaml config (`platform: linux/amd64`)
i will test that on linux later or after merge

**Comment by antdanchenko** [comment]
> I was able to run this on my mac m2, but with some changes in the yaml config (`platform: linux/amd64`) i will test that on linux later or after merge

Thanks, I've made the change to the compose to reflect the platform, according to https://github.com/foundry-rs/foundry/issues/8039#issuecomment-2145539060 that is indeed how people are running the container on Apple Silicon chips this days

btw I can confirm that it runs on
```
  Image: ubuntu-22.04
  Version: 20240526.1.0
  Included Software: https://github.com/actions/runner-images/blob/ubuntu22/20240526.1/images/ubuntu/Ubuntu2204-Readme.md
  Image Release: https://github.com/actions/runner-images/releases/tag/ubuntu22%2F20240526.1
```

**Review by Samyoul — APPROVED** [review]
This looks like really great work. Thank you @antdanchenko !

**Review by Valentina1133 — APPROVED** [review]
All works for me too! Great job @antdanchenko !

**Comment by IvanBelyakoff** [comment]
Great job indeed!

File: integration-tests/tests/test_cases.py [hunk 3]
```diff
@@ -0,0 +1,89 @@
...
+        response = self.rpc_request(method, params, 13)
+        self.verify_is_valid_json_rpc_response(response)
+        return response
+
+    def setup_method(self):
+        super().setup_method()
+        
+        response = self.wallet_create_multi_transaction()
+        try:
+            self.tx_hash = response.json()["result"]["hashes"][str(self.network_id)][0]
```
Comment (yevh-berdnyk) [inline integration-tests/tests/test_cases.py:86]: In current implementation method `wallet_create_multi_transaction()` is called twice for the test `test_create_multi_transaction`. I don't think it's a problem but probably would be good to change this in the future.

File: integration-tests/tests/test_cases.py [hunk 4]
```diff
@@ -0,0 +1,89 @@
+import json
+import requests
+from conftest import option, user_1, user_2
```
Comment (yevh-berdnyk) [inline integration-tests/tests/test_cases.py:3]: Unused imports 
```suggestion
from conftest import option
```

File: integration-tests/tests/test_wallet_rpc.py [hunk 5]
```diff
@@ -0,0 +1,69 @@
...
+
+    def test_create_multi_transaction(self):
+        response = self.wallet_create_multi_transaction()
+        
+        # how to create schema:
+        # from schema_builder import CustomSchemaBuilder
+        # CustomSchemaBuilder("wallet_createMultiTransaction").create_schema(response.json())
+        
+        with open(f"{option.base_dir}/schemas/wallet_createMultiTransaction", "r") as schema:
+            jsonschema.validate(instance=response.json(), schema=json.load(schema))
```
Comment (yevh-berdnyk) [inline integration-tests/tests/test_wallet_rpc.py:47]: I'd suggest to make a separate method for this with a schema name as an input parameter, something like `def validate_json_schema(schema_name)`

**Review by yevh-berdnyk — APPROVED** [review]
Great job! Documentation is really helpful, thanks! Just couple of 'nice to have' comments

File: integration-tests/tests/test_cases.py (hunk 4 above)
Comment (antdanchenko) [reply integration-tests/tests/test_cases.py:3]: they are actually used in transaction data, https://github.com/status-im/status-go/blob/45be54008ed665f4441d5c86ff4b67c4a7b0b631/integration-tests/tests/test_cases.py#L63

**Pull request merged**